examples/checkpointing/checkpointing_example.py

```diff
@@ -1,141 +1,56 @@
 import numpy as np
 from numpy import linalg
 from math import floor
+from cached_property import cached_property
+
 from devito import Function, TimeFunction
 from examples.seismic import Receiver, demo_model, RickerSource
 from examples.seismic.acoustic import ForwardOperator, GradientOperator, smooth10
 from examples.checkpointing.checkpoint import DevitoCheckpoint, CheckpointOperator
+from examples.seismic.acoustic.gradient_example import GradientExample
 from pyrevolve import Revolver
 
+class CheckpointingExample(GradientExample):
+    @cached_property
+    def forward_field(self):
```

Comment:
cached?

Reply:
Making sure that the (memory for the) field is allocated only once in the lifetime of the `CheckpointingExample` object.